src/main/java/athena/TimeAllocator.java

```diff
@@ -23,35 +25,151 @@
+                while (dayLog.get(pos) != -1) {
+                    pos++;
+                    if (pos == sleep) {
+                        done = true;
+                        break;
+                    }
+                }
+                int end = pos;
+                while (dayLog.get(end) == -1) {
+                    end++;
+                    if (end == sleep) {
+                        done = true;
+                        break;
+                    }
+                }
+                int space = end - pos;
+                ArrayList<Integer> temp = new ArrayList<Integer>(Collections.nCopies(space, -1));
+                ArrayList<Task> tasksToRemove = new ArrayList<Task>();
+                while (temp.contains(-1)) {
+                    for (Task currTask : undefinedTimeTasks) {
+                        Integer span = currTask.getTimeInfo().getDuration();
+                        if (span <= space) {
+                            int taskNumber = currTask.getNumber();
+                            this.flexibleTaskList.getTaskFromNumber(taskNumber)
+                                    .getTimeInfo().setStartTime(LocalTime.of(end - space, 0));
+                            for (int i = 0; i < span; i++) {
+                                temp.set(i + end - space - pos, currTask.getNumber());
+                            }
+                            space -= span;
+                            tasksToRemove.add(currTask);
+                            if (space == 0) {
+                                break;
+                            }
+                        }
+                    }
+                    removeAssignedTasks(undefinedTimeTasks, tasksToRemove);
+                    setTempToPerm(dayLog, temp);
+                    if (start == end) {
+                        break;
+                    } else {
+                        start = end;
+                    }
+                    if (undefinedTimeTasks.size() == 0) {
+                        done = true;
+                        break;
+                    }
+                }
+
+            }
+            //System.out.println(fixedDayTasks.getTasks());
+            //System.out.println(flexibleDayTasks.getTasks());
+            currDay.plusDays(1);
+        }
+
+    }
+
+    private ArrayList<Task> getSortedFixedTasks(TaskList taskList) {
+        TaskList fixedDayTasks = taskList;
+        ArrayList<Task> sortedTimeTasks = fixedDayTasks.getTasks();
+        sortedTimeTasks.sort(new TaskTimeComparator());
+        sortedTimeTasks.sort(new TaskImportanceComparator());
+        return sortedTimeTasks;
+    }
+
+    private void removeAssignedTasks(ArrayList<Task> undefinedTimeTasks, ArrayList<Task> tasksToRemove) {
+        for (Task remTask : tasksToRemove) {
+            undefinedTimeTasks.remove(remTask);
+        }
+    }
+
+    private void setTempToPerm(ArrayList<Integer> dayLog, ArrayList<Integer> temp) {
+        int count = 0;
```

Comment:
I thought that `count` would need to start at the starting hour that `temp` started at?